### DISCUSSION TITLE
Custom layout and group by

### DISCUSSION THREAD

**Discussion by buszi2137** [microsoft-search/pnp-modern-search#3252] [General]
Hi, Is it possible to make a grouping of items in my custom layout for search results in PnP Search Results? Have Somebody did this? Could you tell how you made it? I know grouping items works in default list item view in Search Results but we need  to make it run in custom.

**Version used**
4.7

**Desktop:**
 - Chrome Browser

**Comment by wobba** [comment]
It's all about looping the results with handlebars. Look at functions here https://github.com/helpers/handlebars-helpers#array. If you know all possible values, then you can iterate and filter per value as an example. Or use `unique` to find the unique values, then iterate and use that as a filter. I know I've done this in the past but don't have any samples to share unfortunately.
